sympy/matrices/expressions/matmul.py

```diff
@@ -388,8 +389,23 @@ def combine_one_matrices(mul):
 
     return newmul(factor, *new_args)
 
+def distribute_monom(mul):
+    """
+    Simplify MatMul expressions
+
+    e.g. A*(A+B) -> A**2 + A*B
+    """
+    args = mul.args
+    if len(args) == 2:
+        from .matadd import MatAdd
+        if args[0].is_MatAdd:
+            return MatAdd(*[MatMul(mat, args[1]).doit() for mat in args[0].args])
+        if args[1].is_MatAdd:
+            return MatAdd(*[MatMul(args[0], mat).doit() for mat in args[1].args])
```

Comment:
I think that it's better to restrict the expressions that allow distribution
Currently, sympy core only distributes the rational numbers in 2-multiplication, ex) `2*(x + y) -> 2*x + 2*y` and not like `x*(x+y)`, `sqrt(2)*(x+y)`, ...
Maybe we may need to define what 'rational matrix expression' should be, like linear combination of rational numbers and one and zero and identity matrices, but this can be too complicated and I think that it should be sufficient to distribute only rational scalars in 2-multiplication to distribute negative signs.

Sometimes the distributive law isn't desirable for automatic evaluation. As you see in the example of matrix normal distribution, now the expression is way more complicated containing many additive terms, and it should be in the scope of `expand` to have full polynomial expansions.

Reply:
It should be possible to filter out the rational scalar expressions by using `is_Rational`

Reply:
> Currently, sympy core only distributes the rational numbers in 2-multiplication

I don't think that the way distribution happens with `Mul` is a good idea so I don't suggest copying that as a design anywhere else.